### ISSUE TITLE
Pylint doesn't warn if isinstance's second argument doesn't consist entirely of types.

### ISSUE THREAD

**Issue by bnsh** [PyCQA/pylint#3308]
Not really sure if it's a real issue... But, the bug (my own) bit me, and I thought pylint would have saved me from it...

Here's the example code:

```
#! /usr/bin/env python3
# vim: expandtab shiftwidth=4 tabstop=4

"""This program demonstrates that pylint doesn't catch
   instanceof "types" """

def main():
    print(isinstance("x", hash))

if __name__ == "__main__":
    main()
```

If I run it tho, I get:

```
$ python3 ./demo.py 
Traceback (most recent call last):
  File "./demo.py", line 11, in <module>
    main()
  File "./demo.py", line 8, in main
    print(isinstance("x", hash))
TypeError: isinstance() arg 2 must be a type or tuple of types
```

it's pylint clean tho:

```
$ python3 -m pylint --version
pylint 2.4.3
astroid 2.3.2
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0]
$ python3 -m pylint -r n demo.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

```

Is that a problem? (In this case, I said "hash" where I meant "dict")

Thanks!

**Comment by PCManticore** [comment]
Thank you for the report. This is definitely an issue we should be able to detect.

**Comment by anubh-v** [comment]
I'll attempt this.
